onedal/common/tests/test_sycl.py

```diff
@@ -0,0 +1,118 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import pytest
+
+from onedal import _backend, _is_dpc_backend
+from onedal.tests.utils._device_selection import get_queues
+from onedal.utils._dpep_helpers import dpctl_available
+
+
+@pytest.mark.skipif(
+    not _is_dpc_backend or not dpctl_available, reason="requires dpc backend and dpctl"
+)
+@pytest.mark.parametrize("device_type", ["cpu", "gpu"])
+@pytest.mark.parametrize("device_number", [None, 0, 1, 2, 3])
+def test_sycl_queue_string_creation(device_type, device_number):
+    # create devices from strings
+    from dpctl import SyclQueue
+    from dpctl._sycl_queue import SyclQueueCreationError
+
+    onedal_SyclQueue = _backend.SyclQueue
+
+    device = ":".join([device_type, str(device_number)]) if device_number else device_type
+
+    raised_exception_dpctl = False
+    raised_exception_backend = False
+
+    try:
+        dpctl_queue = SyclQueue(device)
+    except SyclQueueCreationError:
+        raised_exception_dpctl = True
+
+    try:
+        onedal_queue = onedal_SyclQueue(device)
+    except RuntimeError:
+        raised_exception_backend = True
+
+    assert raised_exception_dpctl == raised_exception_backend
+    # get_device_id must be modified to follow DPCtl conventions
+    # this causes filter_string mismatches
+    # if not raised_exception_backend:
+    #    assert (
+    #        onedal_queue.sycl_device.filter_string
+    #        in dpctl_queue.sycl_device.filter_string
+    #    )
+
+
+@pytest.mark.skipif(
+    not _is_dpc_backend or not dpctl_available, reason="requires dpc backend and dpctl"
+)
+@pytest.mark.parametrize("queue", get_queues())
+def test_sycl_queue_conversion(queue):
+    if queue is None:
+        pytest.skip("Not a DPCtl queue")
+    SyclQueue = queue.__class__
+    onedal_SyclQueue = _backend.SyclQueue
+    # convert back and forth to test `_get_capsule` attribute
+
+    q = onedal_SyclQueue(queue)
+    # get_device_id must be modified to follow DPCtl conventions
+    # assert q.sycl_device.filter_string in queue.sycl_device.filter_string
```

Comment:
is this meant to be commented? test feels incomplete in current state

Reply:
there is a ticket for the device filter mismatch, it is currently unusued in our codebase with this PR